sound/soc/sof/intel/hda-loader.c

```diff
@@ -372,3 +372,148 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	dev_err(sdev->dev, "error: load fw failed err: %d\n", ret);
 	return ret;
 }
+
+// skl/kbl enable core and code loader DMA has some difference with apl/cnl
+// add the APIs for skl/kbl
+static int cl_stream_prepare_skl(struct snd_sof_dev *sdev, unsigned int format,
+				 unsigned int size, struct snd_dma_buffer *dmab,
+				 int direction)
+{
+	u32 tag = 0;
+	return tag;
+}
+
+static int cl_dsp_init_skl(struct snd_sof_dev *sdev, const void *fwdata,
+			   u32 fwsize)
+{
+	int tag, ret, i;
+	u32 hipcie;
+	u32 reg;
+	const struct sof_intel_dsp_desc *chip = sdev->hda->desc;
```

Comment:
structures should always be at the top of the variables.

Reply:
OK, will change that.